www/templates/html/MediaPlayer.tpl.php

```diff
@@ -6,13 +6,184 @@
+                    var $transcript;
+                    var $captionSearch;
+
+                    <?php if($context->media->getTextTracks()): ?>
+
+                        t.container.append(WDN.jQuery(".mh_transcript_template").html());
+
+                        $transcript = t.container.find('.mh-transcript');
+                        $captionSearch = t.container.children(".mh-caption-search");
+
+                        var $myButton = $('<div>', {
+                            "class": "mejs-button wide"
+                            }).append($('<button>', {
+                                "class": "wdn-icon-search caption-toggle", 
+                                "type": "button",
+                                "aria-controls": t.id,
+                                "title": "Toggle Searchable Transcript",
+                                "aria-label": "Toggle Searchable Transcript"
+                            }));                       
+
+                        t.captionsButton.before($myButton)
+
+                        t.controls.on("click", ".caption-toggle", function(){
+                            $captionSearch.toggleClass("show");
+                        });
+                        var displaytime = function(millis){
+
+                            var hours = Math.floor(millis / 36e5),
+                                mins = Math.floor((millis % 36e5) / 6e4),
+                                secs = Math.floor((millis % 6e4) / 1000);
+
+                                if(secs < 10){
+                                    secs = "0"+secs;
+                                }
+
+                                if(hours > 0){
+                                    return hours+':'+mins+':'+secs;
+                                }else{
+                                    return mins+':'+secs;
+                                };
+
+                        }
+
+                        var setTranscript = function(track){
+
+                            $captionSearch.find(".mh-caption-close").on("click", function(){
+
+                                $(this).siblings(".mh-parse-caption").val("");
+
+                                $transcript.find("li").addClass("highlight");
+
+                            });
+
+                            $captionSearch.find(".mh-parse-caption").on("keyup focus", function(e){
+
+                                e.stopPropagation();
+
+                                var search = $(this).val().toLowerCase();
+                                var subtitlesLength;
+                                var i;
+
+                                //$transcript.find("li").removeClass("highlight");
+                                if (!search) {
+                                    return;
+                                }
+
+                                subtitlesLength = track.entries.text.length;
+
+                                for (i = 0; i < subtitlesLength; i++) {
+
```

Comment:
![I find your overuse of white-space disturbing](https://cloud.githubusercontent.com/assets/675956/8749035/4084abea-2c64-11e5-9356-8d824f66c82b.jpg)